src/main/java/com/untamedears/ItemExchange/utility/BlockUtility.java

```diff
@@ -0,0 +1,111 @@
+package com.untamedears.ItemExchange.utility;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.*;
+import org.bukkit.inventory.*;
+
+import com.untamedears.ItemExchange.DeprecatedMethods;
+import com.untamedears.ItemExchange.ItemExchangePlugin;
+
+public class BlockUtility {
+
+	public static final BlockFace[] CardinalFaces = new BlockFace[] {
+			BlockFace.NORTH,
+			BlockFace.EAST,
+			BlockFace.SOUTH,
+			BlockFace.WEST,
+			BlockFace.DOWN,
+			BlockFace.UP
+	};
+
+	public static BlockFace GetFacingDirection(Block block) {
+		byte data = DeprecatedMethods.getBlockMeta(block);
+		if ((data & 0x5) == 5)
```

Comment:
This is really weird code, you should have just done something like

```
byte facing = data & 0x7;
switch(facing) {
  case 0: 
    return BlockFace.DOWN;
  case 1:
    return BlockFace.UP;
etc.
}
```

Reply:
I don't know binary too well, it would be better for someone more experienced to do that.